starter-core/src/main/java/io/micronaut/starter/feature/discovery/DiscoveryCore.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2017-2023 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.starter.feature.discovery;
+
+import io.micronaut.starter.application.generator.GeneratorContext;
+import io.micronaut.starter.build.dependencies.MicronautDependencyUtils;
+
+public abstract class DiscoveryCore implements DiscoveryFeature {
```

Comment:
I think we will need a non abstract class so that users are able to select a discovery-core even if they don't use consul, k8s or eureka.